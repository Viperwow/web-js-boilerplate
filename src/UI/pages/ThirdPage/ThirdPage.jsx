import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class ThirdPage extends Component {
  render() {
    const {text} = this.props;

    return (
      <div className="page">
        {text}
        <ul>
          <li>
            <Link to="/1" replace>
              1
            </Link>
          </li>
          <li>
            <Link to="/2" replace>
              2
            </Link>
          </li>
          <li>
            <Link to="/not-found" replace>
              4
            </Link>
          </li>
        </ul>
      </div>
    );
  }
}

export default ThirdPage;
