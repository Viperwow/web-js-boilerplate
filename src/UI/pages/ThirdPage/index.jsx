// Vendors
import React, {Component} from 'react';

class ThirdPage extends Component {
  render() {
    const {text} = this.props;

    return (
      <div className="page">
        {text}
      </div>
    );
  }
}

export default ThirdPage;
