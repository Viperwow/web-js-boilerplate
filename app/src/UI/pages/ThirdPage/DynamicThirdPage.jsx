import React from 'react';
import dynamic from 'src/common/dynamic';

const DynamicThirdPage = () => {
  const Component = dynamic(
    import('src/UI/pages/ThirdPage' /* webpackChunkName: "ThirdPage" */),
  );

  return <Component />;
};

export default DynamicThirdPage;
