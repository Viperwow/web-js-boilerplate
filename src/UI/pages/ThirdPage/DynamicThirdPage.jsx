// HoC
import loadable from 'src/common/loadable';

const DynamicThirdPage = loadable(
  import('src/UI/pages/ThirdPage' /* webpackChunkName: "ThirdPage" */),
)();

export default DynamicThirdPage;
