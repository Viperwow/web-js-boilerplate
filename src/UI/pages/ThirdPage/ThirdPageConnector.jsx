// Vendors
import i18n from 'i18next';
// Common
import connectQuery from 'src/common/connectQuery';
// GQL
import QUERY_LOCALE from 'src/queries/locale';
// UI
import ThirdPage from './ThirdPage';

const mapStateToProps = () => ({
  text: i18n.t('pages.third'),
});

export default connectQuery(QUERY_LOCALE, mapStateToProps)(ThirdPage);
