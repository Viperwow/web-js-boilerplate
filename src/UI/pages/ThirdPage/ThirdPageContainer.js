import connectQuery from 'src/common/connectQuery';
import withOnError from 'src/common/withOnError';
import withTranslate from 'src/common/withTranslate';
import {compose} from 'src/helpers/utility';
import {GET_APP_LOCALE} from 'src/queries';
import ThirdPage from './ThirdPage';

const mapStateToProps = (_, {t}) => ({
  text: t('pages.third'),
});

export default compose(
  withTranslate,
  connectQuery(GET_APP_LOCALE, mapStateToProps),
  withOnError,
)(ThirdPage);
