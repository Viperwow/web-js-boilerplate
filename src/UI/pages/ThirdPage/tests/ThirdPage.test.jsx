// Vendors
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
// UI
import ThirdPage from '..';

describe('ThirdPage component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<ThirdPage />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
