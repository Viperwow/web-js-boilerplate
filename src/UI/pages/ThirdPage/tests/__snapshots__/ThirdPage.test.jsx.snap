// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdPage component should render correctly With default props 1`] = `
<div
  className="page"
/>
`;

exports[`ThirdPage component should render correctly With text prop 1`] = `
<div
  className="page"
>
  text
</div>
`;
